src/OpenSearch.Client/QueryDsl/Abstractions/Container/QueryContainer-Dsl.cs

```diff
@@ -131,5 +131,42 @@ public void Accept(IQueryVisitor visitor)
 
 		// ReSharper disable once UnusedMember.Global
 		internal bool ShouldSerialize(IJsonFormatterResolver formatterResolver) => IsWritable;
+
+
+		public QueryContainer Name(string name)
+		{
+			ContainedQuery.Name = name;
+			return this;
+		}
+
+		/// <summary>
+		/// Applies or removes the `strict` attribute to the query container and all child sub-queries.
+		/// </summary>
+		/// <param name="strict">When null, the value is taken from the root query.</param>
+		/// <returns></returns>
+		public QueryContainer Strict(bool? strict = true)
+		{
+			if (!strict.HasValue)
+				strict = ContainedQuery.IsStrict;
+
+			var visitor = new QueryNodeModifierVisitor((query, ctx) => query.IsStrict = strict.Value);
```

Comment:
It may end up being simpler and less code to just make two very short specialised implementations of QueryVisitor that can be class private ie something like `new StrictnessPropogatingVisitor(bool strict)`

Reply:
I usually go for more general-purpose implementations, but your suggestion will definitely solve my problem finding a good name for my current class implementation... ;)